tetrio-plus: split into seperate derivation

huantianad opened the pull request:
## Description of changes
This allows the tetrio-plus package to be cached, so users don't have to compile tpsecore themselves.

@wackbyte let me know if this is something we even want to do. 

Questions:
- Do we want to switch `tetrio-desktop`'s `version` to just be `9`? Upstream seems to only be using integer versions, and it would save us some `lib.versions.major` calls.
- Is it OK to reference `tetrio-desktop`'s `src` and `version` like this in general?

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

huantianad committed:
tetrio-desktop: format with nixfmt

huantianad committed:
tetrio-desktop: add --enable-wayland-ime

wackbyte commented on pkgs/by-name/te/tetrio-plus/package.nix:
```diff
@@ -1,29 +1,30 @@
     owner = "UniQMG";
     repo = "tetrio-plus";
-    rev = "electron-v${version}-tetrio-v${lib.versions.major tetrio-version}";
+    rev = rev;
```
```suggestion
    inherit rev;
```

wackbyte commented:
seems like there's a precedent for this with Vencord and OpenASAR, so LGTM!

> Do we want to switch `tetrio-desktop`'s `version` to just be `9`? Upstream seems to only be using integer versions, and it would save us some `lib.versions.major` calls.

sounds good to me!

> Is it OK to reference `tetrio-desktop`'s `src` and `version` like this in general?

I think so.

huantianad committed:
tetrio-desktop: switch version to single integer to match upstream

Upstream brands each version as "v9" or "Version 9" for example, and
tools like TETR.IO PLUS also use only the major version.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1954

huantianad commented:
@NixOS/nixpkgs-merge-bot merge

SuperSandro2000 commented on pkgs/by-name/te/tetrio-plus/package.nix:
```diff
@@ -148,16 +148,21 @@ stdenv.mkDerivation (finalAttrs: {
-    downloadPage = "https://gitlab.com/UniQMG/tetrio-plus/-/releases";
+    description = "Modified TETR.IO desktop app.asar with many customization tools";
+    longDescription = ''
+      Modified TETR.IO desktop app.asar with the TETR.IO PLUS customization toolkit.
+
```
```suggestion
```

longDescription should not repeat description

SuperSandro2000 commented on pkgs/by-name/te/tetrio-plus/package.nix:
```diff
@@ -14,17 +14,17 @@
     owner = "UniQMG";
     repo = "tetrio-plus";
-    rev = "electron-v${version}-tetrio-v${lib.versions.major tetrio-version}";
+    inherit rev;
```
Why move rev?

huantianad replied:
Used in changelog, wanted to reduce duplication, can revert.

huantianad committed:
tetrio-plus: cleanup and reorganize build phase

huantianad committed:
tetrio-plus: split into seperate derivation

huantianad committed:
tetrio-desktop,tetrio-plus: update meta

huantianad commented:
I've split the 'tetrio-plus: split into seperate derivation' commit into two commits to hopefully make the changes a bit clearer; let me know if some of the commits make sense to be squashed.
